lib/parse-value.js

```diff
@@ -0,0 +1,40 @@
+var util = require('util');
+
+module.exports = function parseValue(value, parseAsJson) {
+	if (parseAsJson) {
+		try {
+			return JSON.parse(value)
+		} catch (error) {
+			throw new Error(
+				util.format(
+					'The received value cannot be parsed as a JSON string. Executed:\n'
+					+ 'JSON.parse(%j).\n'
+					+ 'You might want to wrap the string in single quotes, for example:\n'
+					+ 'dot-json file.json key \'{"value": 42}\' --json-value',
+					value
```

Comment:
The native message by `JSON.parse` is pretty useless and that's compounded by the string handling by the console. I thought it'd be best to show exactly what's being executed and how to fix it

<img width="801" alt="Screenshot" src="https://user-images.githubusercontent.com/1402241/227786866-701e3927-4884-4c14-8fa7-608231bdd727.png">